types/base_app.go

```diff
@@ -7,36 +7,36 @@ func NewBaseApplication() *BaseApplication {
 	return &BaseApplication{}
 }
 
-func (app *BaseApplication) Info(req RequestInfo) (resInfo ResponseInfo) {
-	return
+func (BaseApplication) Info(req RequestInfo) ResponseInfo {
+	return ResponseInfo{}
 }
 
-func (app *BaseApplication) SetOption(key string, value string) (log string) {
+func (BaseApplication) SetOption(key string, value string) (log string) {
 	return ""
 }
 
-func (app *BaseApplication) DeliverTx(tx []byte) Result {
-	return NewResultOK(nil, "")
+func (BaseApplication) DeliverTx(tx []byte) ResponseDeliverTx {
+	return ResponseDeliverTx{Code: CodeType_OK}
```

Comment:
btw, why DeliverTx/CheckTx/Commit return OK, but Query returns empty ResponseQuery{}?

Reply:
LOL good eye. Should be effectively the same since OK is the zero value, but we should be consistent.